com.heroicgameslauncher.hgl.yml

```
@@ -0,0 +1,148 @@
+  - --filesystem=home
+  - --filesystem=xdg-data/Steam:ro
+  - --filesystem=~/.var/app/com.valvesoftware.Steam:ro
+  - --persist=.wine
+  - --require-version=1.1.2
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --talk-name=org.freedesktop.Notifications
+
+add-extensions:
+  org.freedesktop.Platform.Compat.i386:
+    directory: lib/i386-linux-gnu
+    version: '21.08'
+
+  org.freedesktop.Platform.Compat.i386.Debug:
+    directory: lib/debug/lib/i386-linux-gnu
+    version: '21.08'
+    no-autodownload: true
+
+  org.freedesktop.Platform.GL32:
+    directory: lib/i386-linux-gnu/GL
+    version: '1.4'
+    versions: 21.08;1.4
+    subdirectories: true
+    no-autodownload: true
+    autodelete: false
+    add-ld-path: lib
+    merge-dirs: vulkan/icd.d;glvnd/egl_vendor.d;OpenCL/vendors;lib/dri;lib/d3d;vulkan/explicit_layer.d;vulkan/implicit_layer.d
+    download-if: active-gl-driver
+    enable-if: active-gl-driver
+
+modules:
+
+  # --- Tools ---
+  - name: unappimage
+    buildsystem: simple
+    build-commands:
+      - make -C squashfs-tools -j ${FLATPAK_BUILDER_N_JOBS} install INSTALL_DIR=/app/bin
+    sources:
+      - type: git
+        url: https://github.com/refi64/unappimage
+        commit: d7f86f2a0d7ec3a69211125207d5f127386b849a
+
+  - name: vulkan-tools
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DGLSLANG_INSTALL_DIR=/app
+      - -DVULKAN_HEADERS_INSTALL_DIR=/app
+      - -DCMAKE_BUILD_TYPE=Release
+    sources:
+      - type: archive
+        url: https://github.com/KhronosGroup/Vulkan-Tools/archive/refs/tags/v1.2.186.tar.gz
+        sha256: 8ba0e21e8de0b70d31985d251d11150efee71266875262798784e9d1ec9b717b
+
+  - name: gamemode
+    buildsystem: meson
+    config-opts: &gamemode_opts
+      - -Dwith-systemd=false
+      - -Dwith-daemon=false
+      - -Dwith-examples=false
+      - -Dwith-util=false
+      - -Dwith-sd-bus-provider=no-daemon
+    sources: &gamemode_sources
+      - type: archive
+        url: https://github.com/FeralInteractive/gamemode/releases/download/1.6.1/gamemode-1.6.1.tar.xz
+        sha256: 10c2a3f142eae472f5a09e42616e38c666c05b25ca3e61e562a543bb3fda66c5
+
+  - name: gamemode-32bit
+    build-options:
+      arch:
+        x86_64: &compat_i386_opts
+    buildsystem: meson
+    config-opts: *gamemode_opts
+    sources: *gamemode_sources
+
+  - name: gamemoderun
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 data/gamemoderun -t /app/bin
+    sources: *gamemode_sources
+
+  # --- Heroic ---
+  - name: heroic
+    buildsystem: simple
+    build-commands:
+      - unappimage Heroic-2.2.2.AppImage
+      - rm Heroic-2.2.2.AppImage
+      - mv squashfs-root /app/bin/heroic
+      - install -D heroic-run -t /app/bin
+    sources:
+      - type: script
+        dest-filename: heroic-run
+        commands:
+          - zypak-wrapper /app/bin/heroic/heroic "$@"
+
+      - type: file
+        filename: Heroic-2.2.2.AppImage
```

Consider configuring https://github.com/flathub/flatpak-external-data-checker/ (see GitHub API example in README), unless you plan to automate it somehow else/maintain it fully manually.

---

I will check this out.

---

Basically, add `x-checker-data` object property
```suggestion
        filename: Heroic-2.2.2.AppImage
        x-checker-data:
          type: electron-updater
          url: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/releases/latest/download/latest-linux.yml
```
And change `sha256` to `sha512`

---

Yes, thats one me because I found a bug on Heroic after the release that was affecting the Flatpak, so I fixed it and uploaded a new AppImage.
But this is not common to happen and I will release a new version of Heroic today with the fixes.

---

This is not a blocker, just the update checker will not actually submit updates until the metadata is fixed or a new release comes out.